vic/drivers/image/src/get_force_type.c

```diff
@@ -0,0 +1,148 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * This routine determines the current forcing file data type and stores its
+ * location in the description of the current forcing file.
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2014 The Land Surface Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <vic_def.h>
+#include <vic_run.h>
+#include <vic_driver_image.h>
+
+/******************************************************************************
+ * @brief    This routine determines the current forcing file data type and
+ *           stores its location in the description of the current forcing file.
+ *****************************************************************************/
+void
+get_force_type(char *cmdstr,
```

Comment:
It's still a poor interface even if we defined it that way before.  So if we keep it that way for historic reasons that is fine, but a function named `get_force_type` should just do that or be renamed. There is no argument that suddenly makes that a good interface. Since `vic_classic` has its own separate driver code base (i.e. this is not part of the `shared` library), there is no reason they have to remain aligned.

So either:
a. Leave as is for historic reasons (and state that - probably include a comment in `get_force_type` to make that clear;

or

b. Fix it.